src/periodic.jl

```diff
@@ -0,0 +1,85 @@
+const PeriodicFunction = Union{typeof(cos), typeof(sin)}
+const PeriodicParameter = Union{Real,Period}
+
+"""
+    Periodic(f, period, phase_shift) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+# Fields
+* `f::PeriodicFunction`: the periodic function
+* `period<:PeriodicParameter`: the duration it takes for the periodic function to repeat.
+    The sign of the value determines the direction of the function. Must be non-zero.
+* `phase_shift<:PeriodicParameter`: adjusts the phase of the periodic function, measured
+    in the same units as the input. Increasing the value translates the function to the
+    right, toward higher/later input values.
+"""
+struct Periodic{T<:PeriodicParameter} <: Transform
+    f::PeriodicFunction
+    period::T
+    phase_shift::T
+end
+
+"""
+    Periodic(f::PeriodicFunction, period<:PeriodicParameter) -> Periodic
+
+A constructor for [`Periodic`](@ref).
+Returns a `Periodic` transform with zero phase shift.
+"""
+function Periodic(f, period)
+    return Periodic{typeof(period)}(f, period, zero(typeof(period)))
+end
+
+function _apply!(x::AbstractArray{T}, P::Periodic; kwargs...) where T <: Real
+    x[:] = P.f.(2π .* (x .- P.phase_shift) / P.period)
+    return x
+end
+
+function apply(x::AbstractArray{T}, P::Periodic{U}; kwargs...) where {T <: TimeType, U <: Period}
+    return map(xi -> _periodic(P.f, xi, P.period, P.phase_shift), x)
+end
+
+"""
+    Transforms.apply(table, ::Periodic{T}; cols=nothing) where T <: Period -> Array
+
+Applies [`Periodic`](@ref) to each of the specified columns in `table`.
+If no `cols` are specified, then [`Periodic`](@ref) is applied to all columns.
+Returns an array containing each transformed column.
+"""
+function apply(table, P::Periodic{T}; cols=nothing) where T <: Period
+    Tables.istable(table) || throw(MethodError(apply, (table, P)))
+
+    columntable = Tables.columns(table)
+    cnames = cols === nothing ? propertynames(columntable) : cols
+    return [apply(getproperty(columntable, cname), P) for cname in cnames]
```

Comment:
Rethinking this - should it return a DataFrame or NamedTuple just to give the columns names, even though this method is agnostic to the specific table type? @nicoleepp

Reply:
It should return a table - since an array isn't expected in this instance - so maybe a NamedTuple with the same column names is appropriate?

Reply:
Yeah makes sense. I created a broader issue about this #12. It also raises the question of transformed column naming, but I can use the same column names for now.

Reply:
although look at the same function in https://github.com/invenia/Transforms.jl/pull/8

https://github.com/invenia/Transforms.jl/pull/8/files#diff-e6cb55b7b445a33fcc97a850d6c3907c75fdd150531bf7b42f04aae186b9b869R44-R54

which sums the data from a collection of cols, how would we name the return in this instance? 

It starts to get confusing/complicated when you consider the types of transformations we have (one-to-one, many-to-one, etc.) and trying to come up with rules that apply to all.

So maybe reconsider my suggestion for a NamedTuple for now. What do you think @nicoleepp ?

Reply:
For the record I'm keeping this as-is.

Reply:
A note on this, it's not trivial to construct a new table of the same type as the old one (aka the way to do this would be different for every table AFAIK) so just returning arrays helps with this too